pkg/storage/snapshot/restore.go

```diff
@@ -596,6 +606,125 @@ func (t *vmRestoreTarget) reconcileDataVolumes() (bool, error) {
 	return createdDV || waitingDV, nil
 }
 
+func isControllerRevisionOwnedByObject(cr *appsv1.ControllerRevision, objMeta metav1.ObjectMeta) bool {
+	for _, ownerRef := range cr.OwnerReferences {
```

Comment:
This should actually use [`metav1.GetControllerOf`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#GetControllerOf)